service/modification/Refund.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen PrestaShop plugin
+ *
+ * Copyright (c) 2019 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ */
+
+namespace Adyen\PrestaShop\service\modification;
+
+use AbstractLogger;
+use Adyen\AdyenException;
+use Adyen\Service\Modification;
+use Adyen\Util\Currency;
+use Db;
+use OrderSlip;
+use PrestaShopDatabaseException;
+
+class Refund
+{
+    /**
+     * @var Modification
+     */
+    private $modificationClient;
+
+    /**
+     * @var string
+     */
+    private $merchantAccount;
+    /**
+     * @var Db
+     */
+    private $db;
+    /**
+     * @var AbstractLogger
+     */
+    private $logger;
+
+    /**
+     * Refund constructor.
+     * @param Modification $modificationClient
+     * @param Db $db
+     * @param $merchantAccount
+     * @param AbstractLogger $logger
+     */
+    public function __construct(
+        Modification $modificationClient,
+        Db $db,
+        $merchantAccount,
+        AbstractLogger $logger = null
+    ) {
+        $this->modificationClient = $modificationClient;
+        $this->db = $db;
+        $this->merchantAccount = $merchantAccount;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param string $orderId
+     * @param string $amount
+     * @param string $currency
+     * @return bool
+     * @throws AdyenException
+     * @throws PrestaShopDatabaseException
+     */
+    public function request($orderId, OrderSlip $orderSlip, $currency)
+    {
+        $amount = (new Currency())->sanitize($orderSlip->amount, $currency);
+
+        $orderId = pSQL($orderId);
+        $results = $this->db->executeS(sprintf(
```

Comment:
Because multi-line strings are a pain to deal with, so using a Heredoc makes the query itself easier to read while keeping the file structure somewhat sane. And then Heredocs don't support concatenation, so `sprintf` deals with that.